docs: add supported technolgies page (#137)
elastic/apm-agent-js-base: docs/supported-technologies.asciidoc

[[supported-technologies]]
== Supported Technologies

> The Elastic APM RUM JavaScript agent will automatically instrument the following to give you detailed performance metrics of your web page

Reviewer: ```suggestion
The Elastic APM RUM JavaScript agent gives you detailed performance metrics of your web page by automatically instrumenting the following:
```

* Page load metrics
* Load time of Static Assets
* API requests (XMLHttpRequest and Fetch)

The agent uses the https://developer.mozilla.org/en-US/docs/Web/API/Navigation_timing_API[Navigation Timing API] and [Resource Timing API] available in the browsers to instrument the page load performance and static assets load times.

> The agent uses <<custom-transactions, Custom Transactions API>> to instrument all the outgoing XHR and Fetch API requests from the web page to the backend servers and also adds support for <<distributed-tracing-guide, Distributed Tracing>>.

Reviewer 2: I think we should say something like: `The agent automatically captures all the outgoing http requests, by instrumenting both XHR and Fetch APIs ....`

Reviewer: Should probably also break distributed tracing into its own sentence to avoid a run on sentence

[float]
[[frameworks]]
=== Frameworks

> The agent is not framework agnostic and therefore will not instrument metrics related to rendering time or mounting time of components. If you are interested in measuring custom timings on frameworks like React, Angular, vue etc., please use the <<apm-start-span, Span API>> and create custom spans instead.

Author: damm i screwed that up 😄 totally
